WordPress/Classes/ViewRelated/Blog/Site Settings/HomepageSettingsViewController.swift

```diff
@@ -118,37 +144,79 @@ import WordPressShared
             postsPageTitle = postsPage.titleForDisplay()
         }
 
-        let selectPage: ImmuTableAction = { _ in
+        let homepageRow: ImmuTableRow
+        if case .selectedHomepage = updateState {
+            homepageRow = ActivityIndicatorRow(title: Strings.homepage, animating: true, action: nil)
+        } else {
+            homepageRow = NavigationItemRow(title: Strings.homepage, detail: homepageTitle, action: { _ in
+                self.pushPageSelection(selectedPostID: self.blog?.homepagePageID) { [weak self] selected in
```

Comment:
Could we extract the contents of this `action` to another method or something to make it a little clearer to follow? Also, should this self be weak?